### PR TITLE
dotCMS/core#27536 Creation of a Centralized, Buildable Experiment Logic Library in Nx

### LINKED ISSUE #27536: Creation of a Centralized, Buildable Experiment Logic Library in Nx
### User Story
As a developer, I want to create a buildable and publishable Nx library to centralize the experiment logic, ensuring consistency and reusability across various parts of the application.

### Acceptance Criteria
- A new Nx library for experiment logic is created.
- The library should be buildable, allowing it to be compiled separately from the applications.
- Centralized experiment logic should encapsulate all functionalities related to experiments in the frontend.
- Proper documentation on how to use the library within the Nx workspace should be provided.

### Proposed Objective
Core Features

### Proposed Priority
Priority 2 - Important

### dotCMS Version
master

### Quality Assurance Notes & Workarounds
Quality assurance should ensure that the library's functionality is fully covered by tests and that integration with existing applications does not introduce regressions.

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* An empty new Nx Javascript Library for experiment logic is created.

### CODE REVIEW

File: core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Represents an SDK Experiment.
+ * @class
+ */
+export class sdkExperiment {
```
Comment (zJaaal) [inline core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts:5]: use PascalCase for class names

**Review by zJaaal — CHANGES_REQUESTED** [review]

File: core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts (hunk 1 above)
Comment (oidacra) [reply core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts:5]: good catch. Let me check if I can setup `eslint` to force that always.

File: core-web/libs/sdk/experiments/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "sdk-experiments",
```
Comment (fmontes) [inline core-web/libs/sdk/experiments/package.json:2]:
```suggestion
  "name": "@dotcms/experiments",
```

File: core-web/package.json [hunk 3]
```diff
@@ -1,259 +1,265 @@
...
-        "ts-jest": "29.1.1",
-        "ts-node": "10.9.1",
-        "tslint": "~6.1.3",
-        "tslint-angular": "^3.0.3",
-        "typedoc": "^0.25.4",
-        "typescript": "~5.2.2",
-        "webpack": "^5.64.0",
-        "webpack-bundle-analyzer": "^4.5.0"
-    }
+  "name": "dotcms-ui",
```
Comment (fmontes) [inline core-web/package.json:2]: What happened here? looks like we have some lint or prettier issues.

File: core-web/tsconfig.base.json [hunk 4]
```diff
@@ -1,61 +1,72 @@
...
-            "@models/*": ["apps/dotcms-ui/src/app/shared/models/*"],
-            "@pipes/*": ["apps/dotcms-ui/src/app/view/pipes/*"],
-            "@portlets/*": ["apps/dotcms-ui/src/app/portlets/*"],
-            "@services/*": ["apps/dotcms-ui/src/app/api/services/*"],
-            "@shared/*": ["apps/dotcms-ui/src/app/shared/*"],
-            "@tests/*": ["apps/dotcms-ui/src/app/test/*"]
-        }
-    },
-    "exclude": ["node_modules", "tmp"]
+  "compileOnSave": false,
```
Comment (fmontes) [inline core-web/tsconfig.base.json:2]: Same, we have some linting issues here.

**Review by fmontes — DISMISSED** [review]

File: core-web/libs/sdk/experiments/package.json (hunk 2 above)
Comment (oidacra) [reply core-web/libs/sdk/experiments/package.json:2]: ohh, I followed the sdk-client... I saw now the other 2 use that pattern. I will change it.

Comment (fmontes) [reply core-web/libs/sdk/experiments/package.json:2]: This pattern is what will do that when this library gets published it goes into our npm organization

Comment (oidacra) [reply core-web/libs/sdk/experiments/package.json:2]: Fixed.

File: core-web/tsconfig.base.json (hunk 4 above)
Comment (oidacra) [reply core-web/tsconfig.base.json:2]: let me check.

File: core-web/nx.json [hunk 5]
```diff
@@ -1,132 +1,138 @@
...
-                "ci": {
-                    "ci": true,
-                    "codeCoverage": true
-                }
-            }
-        }
+    "@nrwl/angular:application": {
+      "style": "scss",
+      "linter": "eslint",
+      "unitTestRunner": "karma",
```
Comment (zJaaal) [inline core-web/nx.json:55]: should this be jest?

File: core-web/nx.json [hunk 6]
```diff
@@ -1,132 +1,138 @@
...
-            "!{projectRoot}/.storybook/**/*",
-            "!{projectRoot}/**/*.stories.@(js|jsx|ts|tsx|mdx)",
-            "!{projectRoot}/.storybook/**/*",
-            "!{projectRoot}/**/*.stories.@(js|jsx|ts|tsx|mdx)",
-            "!{projectRoot}/tsconfig.storybook.json",
-            "!{projectRoot}/src/test-setup.[jt]s"
-        ]
+    "@nrwl/angular:library": {
+      "linter": "eslint",
+      "unitTestRunner": "karma"
```
Comment (zJaaal) [inline core-web/nx.json:60]: same

**Review by zJaaal — DISMISSED** [review]

File: core-web/nx.json (hunk 5 above)
Comment (zJaaal) [reply core-web/nx.json:55]: I'm not sure what is `@nrwl` thought, maybe we dont need to change this.

Comment (oidacra) [reply core-web/nx.json:55]: That section of that file, is for set the defaults when you create a new (in this case) application

File: core-web/package.json (hunk 3 above)
Comment (oidacra) [reply core-web/package.json:2]: Fixed was the format:write at the moment of commit. Fixed added at the root properties to the prettier config file.

File: core-web/.prettierrc [hunk 7]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "printWidth": 120
```
Comment (fmontes) [inline core-web/.prettierrc:3]: Does this fix the indenting issue?

File: core-web/tsconfig.base.json (hunk 4 above)
Comment (fmontes) [reply core-web/tsconfig.base.json:2]: Still here.

**Review by fmontes — DISMISSED** [review]

File: core-web/.prettierrc (hunk 7 above)
Comment (oidacra) [reply core-web/.prettierrc:3]: Yes. `format:write` use prettier maybe the default is 80, I put 120 and did not touch the file.

**Review by zJaaal — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Pull request merged**